src/WooCommerce.php

```diff
@@ -52,15 +52,30 @@ public static function woocommerce_add_to_cart($cart_item_key, $product_id, $qua
    * @implements woocommerce_before_single_product
    */
   public static function woocommerce_before_single_product() {
-    if (!static::$isAddedToCart || !$related_accessories_ids = static::getRelatedAccessoriesIds()) {
+    if (!static::$isAddedToCart && get_option('woocommerce_enable_ajax_add_to_cart') !== 'yes') {
```

Comment:
We lost the condition to not output related accessories if none are configured for the product. We don't want to render an empty list of recommendations.

The exit conditions here could use some documentation, because it is really hard to understand the different cases now.  I assume the following, but might be wrong:

```suggestion
    // Only output related accessories after adding it to the cart.
    // If add-to-cart is using Ajax, always output the related accessories
    // on the page as they will be revealed via JavaScript.
    if (!static::$isAddedToCart && get_option('woocommerce_enable_ajax_add_to_cart') !== 'yes') {
      return;
    }
    // Omit related accessories output if none are set for the product.
    if (!$related_accessories_ids = static::getRelatedAccessoriesIds()) {
```